### PR TITLE
Accept key value delimiter as part of value

### PR DESCRIPTION
**Pull request by arthurpassos**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improve the "best-effort" parsing logic to accept `key_value_delimiter` as a valid part of the value. This also simplifies branching and might even speed up things a bit.

### Documentation entry for user-facing changes
Improve the "best-effort" parsing logic to accept `key_value_delimiter` as a valid part of the value. This also simplifies branching and might even speed up things a bit.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02499_extract_key_value_pairs_multiple_input.sql [hunk 1]
```diff
@@ -123,7 +123,7 @@ SELECT
 -- semi-colon as pair delimiter
 -- expected output: {'age':'31','name':'neymar','team':'psg'}
 WITH
-    extractKeyValuePairs('name:neymar;age:31;team:psg;invalid1:invalid1,invalid2:invalid2', ':', ';') AS s_map,
+    extractKeyValuePairs('name:neymar;age:31;team:psg;random_key:value_with_comma,still_part_of_value:still_part_of_value', ':', ';') AS s_map,
```
Comment (Enmk) [inline tests/queries/0_stateless/02499_extract_key_value_pairs_multiple_input.sql:126]: We need some extra test cases, when the value with delimiters are not at the end of input:
`extractKeyValuePairs('name:neymar;age:31;team:psg;random_key:value_with_comma,still_part_of_value:still_part_of_value;anotherkey:anothervalue', ':', ';')`

**Review by Enmk — APPROVED** [review]
LGTM

**Review by vdimir — APPROVED** [review]

**Comment by arthurpassos** [comment]
@vdimir does CI need a re-run? Apparently, the failures are not related, but it does not seem like it'll restart itself

**Comment by arthurpassos** [comment]
Hi @vdimir. I couldn't pin-point if the CI failures (Stress test (debug) and Upgrade check (debug)) are related to the PR, but I am under the impression they are not. If not, is there anything pending for this to be merged?

**Pull request merged**